cpp/src/strings/regex/regcomp.cpp

```diff
@@ -284,27 +324,30 @@ class regex_parser {
             break;
           }
           case 'w':
-            builtins |= cclass_w.builtins;
-            std::tie(is_quoted, chr) = next_char();
-            continue;
-          case 's':
-            builtins |= cclass_s.builtins;
-            std::tie(is_quoted, chr) = next_char();
-            continue;
-          case 'd':
-            builtins |= cclass_d.builtins;
-            std::tie(is_quoted, chr) = next_char();
-            continue;
           case 'W':
-            builtins |= cclass_W.builtins;
+            if (is_ascii(_flags)) {
+              add_ascii_word_class(ranges, chr == 'W');
+            } else {
+              builtins |= (chr == 'w' ? cclass_w.builtins : cclass_W.builtins);
+            }
             std::tie(is_quoted, chr) = next_char();
             continue;
+          case 's':
           case 'S':
-            builtins |= cclass_S.builtins;
+            if (is_ascii(_flags)) {
+              add_ascii_space_class(ranges, chr == 'S');
+            } else {
+              builtins |= (chr == 's' ? cclass_s.builtins : cclass_S.builtins);
```

Comment:
Very minor nit: For 'W' and 'S' a reader might get confused since you compare to uppercase in the `if` block but you compare to lowercase in the `else` block. Using the same comparison (and maybe doing it outside the conditional and saving the result) might avoid any confusion. Obviously a subjective suggestion though.

Reply:
I'm not sure how I feel about this. The main usecase is the lower-case shortcuts. So in the first `chr == <upper-case>` it is asking the function for it's negative. And making the most common usecase for the ternary be the _then_ instead of the _else_ seems more natural. So both of these seem wrong to me:
```
            if (is_ascii(_flags)) {
              add_ascii_space_class(ranges, chr == 'S');
            } else {
              builtins |= (chr == 'S' ? cclass_S.builtins : cclass_s.builtins);
            }
```
```
            if (is_ascii(_flags)) {
              add_ascii_space_class(ranges, chr != 's');
            } else {
              builtins |= (chr == 's' ? cclass_s.builtins : cclass_S.builtins);
            }

```

Reply:
Yeah I see your point. I didn't appreciate that the lowercase is the more common. I'm fine with keeping it as is. I think it's just a mild potential source of confusion, but the tradeoff doesn't seem clearly worth it here.